Feat/elevator

Kredcool opened the pull request:
very basic elevator stuff, before we merge we should:

- [x] update PID
- [x] add preset points
- [x] add controller input
- [x] check declaration of encoder (we may be using rev instead of ctre)

Commit:
elevator without pid (that is coming, don't kill me)

Commit:
unfinished pid, need to add logic to execute and set motor2 as follower

Commit:
fill out this code later

Commit:
adding a encoder instead of the base encoder

Commit:
moving into rio

dublUayaychtee committed:
add default tests

KenwoodFox reviewed (changes requested):
Hey! This is clearly based on `feat/magicbot` and NOT `main` dont try to overmerge!
@dublUayaychtee you need to merge #2 this is the exact issue i warned about in my last review!

@Kredcool you dont support the fixes in #3 here, you'll need to wait for dylan to merge them or include them yourself in this pr.

Commit:
Revert "moving into rio"

This reverts commit 4bba2bdef26589004d85ad054172ada59387eafc.

Commit:
updating workflow for this year

Commit:
fixing regular tests

KenwoodFox commented:
No! This STILL conflicts! @Kredcool check discord! You guys are gonna turn my hair grey :face_holding_back_tears:

Commit:
toggleOnFalse() may be outdated, is being looked into

Commit:
something is borken, check sim errors

Commit:
realized that I may be very stupid, fixed so we can sim again

Commit:
the toggleOnFalse() function was only for commands, replaced with a boolean. Magicbot appears to have no equivalent

Commit:
adding self. and fixing error with calulate()

Commit:
updating code to use proper parts

CoolDude20221 committed:
Added PID constants

Are most probably not accurate.

CoolDude20221 committed:
Updated PID

Variable values set for PID's so only change to PID's will be on elevator constants.

KenwoodFox committed:
Merge branch 'main' into feat/elevator

KenwoodFox committed:
Fix lock

KenwoodFox committed:
Bump userid

KenwoodFox committed:
@code team correctly, lol

KenwoodFox committed:
Fix bad import blocking elevator

KenwoodFox committed:
Boilerplate elevator

KenwoodFox committed:
Add tests, elevator bits et all

KenwoodFox committed:
Discard changes to simgui.json

dublUayaychtee committed:
add make todo function from the other branch

KenwoodFox committed:
Respect manual override

KenwoodFox committed:
Merge pull request #14 from FRC-1721/joe/elevator

KenwoodFox committed:
Fix todo's

KenwoodFox committed:
Add basic setpoints
